pyproject.toml

```diff
@@ -55,9 +55,12 @@ dev = [
     'typing_extensions >= 3.10, < 5; python_version<"3.12"',
     # Unpinned dependencies, latest is fine
     "botocore",
-    "pytest",
+    "boto3",
+    "ovh",
+    'pytest<9',  # Last version with support for Python 3.8 - mypy would complain about the pattern matching otherwise
```

Comment:
Last run on main was on 8.4.2 and green [here](https://github.com/Aiven-Open/rohmu/actions/runs/18974706521/job/54191037016#step:4:349). Without pinning it picked 9.0.x and failed in this PR:

```
.../venv/lib64/python3.13/site-packages/_pytest/terminal.py:1736: error: Pattern matching is only supported in Python 3.10 and greater  [syntax]                                                                                               
```

Bumping the minimum requirement to 3.10 would fix this but would involve bigger CI and code changes to fix other typing issues.